utcp.c

```diff
@@ -26,9 +26,12 @@
 #include <stdint.h>
 #include <stdbool.h>
 #include <string.h>
+
+#ifndef _WIN32
 #include <unistd.h>
 #include <sys/time.h>
 #include <sys/socket.h>
+#endif
```

Comment:
Das hatte ich mir kurz nach meinem Kommentar auch überlegt. Dann wird in der `compat.h` als einziger Stelle diese Weiche nötig und das finde ich auch sehr sinnvoll um den Code übersichtlich zu halten.